### PR TITLE
Improve the gettingStarted example

### PR DESCRIPTION
**Pull request by dschwabeS11**
- Make it easier to use
- Improve resource naming

### CODE REVIEW

File: gettingStarted/sysElevenStackKickstart.yaml [hunk 1]
```diff
@@ -16,30 +16,44 @@ description: Simple template to deploy a single compute instance
 
 parameters:
  key_name:
-  type: string
+   type: string
+   default:
```
Comment (stefreak) [inline gettingStarted/sysElevenStackKickstart.yaml]: Why did you add a default? Without key_name the stack won't work.

File: gettingStarted/sysElevenStackKickstart.yaml [hunk 2]
```diff
@@ -16,30 +16,44 @@ description: Simple template to deploy a single compute instance
 
 parameters:
  key_name:
-  type: string
+   type: string
+   default: 
+ username:
+   type: string
+   default: syseleven
```
Comment (stefreak) [inline gettingStarted/sysElevenStackKickstart.yaml]: I think that makes the stack more complicated and does not add real value

File: gettingStarted/sysElevenStackKickstart.yaml [hunk 3]
```diff
@@ -16,30 +16,44 @@ description: Simple template to deploy a single compute instance
...
+ instance_name:
+   type: string
+   default: kickstart
```
Comment (stefreak) [inline gettingStarted/sysElevenStackKickstart.yaml]: Really good idea! I like it a lot.

I would not parametrize this, but instead use `OS::stack_name` https://docs.openstack.org/heat/latest/template_guide/hot_spec.html#pseudo-parameters

File: gettingStarted/sysElevenStackKickstart.yaml [hunk 4]
```diff
@@ -16,30 +16,44 @@ description: Simple template to deploy a single compute instance
...
+        list_join: [ '-', [ { get_param: instance_name }, 'allow incoming traffic, tcp port 22 and icmp']]
       rules: 
         - { direction: ingress, remote_ip_prefix: 0.0.0.0/0, port_range_min: 22, port_range_max: 22, protocol: tcp } 
         - { direction: ingress, remote_ip_prefix: 0.0.0.0/0, protocol: icmp } 
 
   host:
     type: OS::Nova::Server
     properties:
-      name: kickstart
+      name: {get_param: instance_name }
```
Comment (stefreak) [inline gettingStarted/sysElevenStackKickstart.yaml]: Let's say the customer called the stack "test", it would be nice if the server is called "test-kickstart" in the end

File: gettingStarted/sysElevenStackKickstart.yaml [hunk 5]
```diff
@@ -16,30 +16,44 @@ description: Simple template to deploy a single compute instance
...
       user_data_format: RAW
       user_data: { get_resource: cloud-init-config }    
-      image: Ubuntu Server 16.04 LTS 
-      key_name: {get_param: key_name}
-      flavor: m1.micro
+      image: { get_param: image }
+      key_name: { get_param: key_name }
+      flavor: { get_param: flavor }
```
Comment (stefreak) [inline gettingStarted/sysElevenStackKickstart.yaml]: I like!

File: gettingStarted/sysElevenStackKickstart.yaml [hunk 6]
```diff
@@ -50,28 +64,29 @@ resources:
...
-               UserKnownHostsFile=/dev/null
-               StrictHostKeyChecking=no
-              EOF
-              chown -R syseleven: /home/syseleven
+              until id ${USERNAME} > /dev/null; do sleep 2; done
+              echo "Found id: ${USERNAME}"
+              mkdir /home/${USERNAME}/.ssh
+              # create empty file for next command not to run into an error
+              touch /home/${USERNAME}/.ssh/authorized_keys
+              curl -s http://169.254.169.254/latest/meta-data/public-keys/0/openssh-key >> /home/${USERNAME}/.ssh/authorized_keys
```
Comment (stefreak) [inline gettingStarted/sysElevenStackKickstart.yaml]: I think parametrizing the username does not add enough value to justify the added complexity.

File: gettingStarted/sysElevenStackKickstart.yaml [hunk 7]
```diff
@@ -82,13 +97,16 @@ resources:
               export OS_PROJECT_ID="demo_project_id"
               export OS_PASSWORD="demo_password"
 
-              # the following lines don't need to be changed
+              # set other region in NAME and URL if required
+              # available regions: cbk, dbl
```
Comment (stefreak) [inline gettingStarted/sysElevenStackKickstart.yaml]: Let's not mention DBL before it goes live.

File: gettingStarted/sysElevenStackKickstart.yaml [hunk 8]
```diff
@@ -141,13 +162,15 @@ resources:
...
+      name: 
+        list_join: [ '-', [ { get_param: instance_name }, 'net']]
 
   subnet:
     type: OS::Neutron::Subnet
     depends_on: [ net ]
     properties:
-      name: kickstart-subnet
+      name: 
+        list_join: [ '-', [ { get_param: instance_name }, 'subnet']]
```
Comment (stefreak) [inline gettingStarted/sysElevenStackKickstart.yaml]: would be nice if these resources would be called `$stackname-kickstart-subnet` etc.

File: gettingStarted/sysElevenStackKickstart.yaml (hunk 1 above)
Comment (dschwabeS11) [reply gettingStarted/sysElevenStackKickstart.yaml]: I forgot to remove that, used a default in the script before.

File: gettingStarted/sysElevenStackKickstart.yaml (hunk 3 above)
Comment (dschwabeS11) [reply gettingStarted/sysElevenStackKickstart.yaml]: See commit 55e78290a0cba68dfb6b7c3248faa0cdf294eb92

File: gettingStarted/sysElevenStackKickstart.yaml (hunk 4 above)
Comment (dschwabeS11) [reply gettingStarted/sysElevenStackKickstart.yaml]: see commit 55e78290a0cba68dfb6b7c3248faa0cdf294eb92

File: gettingStarted/sysElevenStackKickstart.yaml (hunk 5 above)
Comment (dschwabeS11) [reply gettingStarted/sysElevenStackKickstart.yaml]: me 2

File: gettingStarted/sysElevenStackKickstart.yaml (hunk 7 above)
Comment (dschwabeS11) [reply gettingStarted/sysElevenStackKickstart.yaml]: removed
